app/controllers/tenants/projects_controller.rb

```diff
@@ -26,10 +26,20 @@ def update
     @project = Project.find(params[:id])
     if @project.update_attributes(new_params)
       flash[:notice] = "#{@project.title} Updated"
+      if session[:admin]
+        redirect_to admin_projects_path
+      else
+        redirect_to tenant_dashboard_path
+      end
     else
-      flash[:errors] = "Invalid Attributes"
+      if new_params[:categories].empty?
+        flash[:errors] = "Please select one or more categories"
+        redirect_to edit_tenant_project_path(@project, slug: @project.tenant.id)
+      else
+        flash[:errors] = "Invalid Attributes"
+        redirect_to edit_tenant_project_path(@project, slug: @project.tenant.id)
```

Comment:
In both branches, the redirect_to goes to the same place, so put it after the end of the if/else.